scanner/src/main/kotlin/RemoteScanner.kt

```diff
@@ -0,0 +1,37 @@
+/*
+ * Copyright (C) 2020 Bosch.IO GmbH
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ * License-Filename: LICENSE
+ */
+
+package org.ossreviewtoolkit.scanner
+
+import org.ossreviewtoolkit.model.ScannerDetails
+import org.ossreviewtoolkit.model.config.ScannerConfiguration
+
+/**
+ * Abstraction for a [Scanner] that runs on a remote host.
+ */
+abstract class RemoteScanner(name: String, config: ScannerConfiguration) : Scanner(name, config) {
+    abstract fun getVersion(): String
+
+    abstract fun getConfiguration(): String
```

Comment:
Maybe this is a bit out of scope, but nevertheless: The scanner version and configuration are typically static. So it could make sense to declare them as abstract properties. This would allow some optimizations, e.g. the scanner details could then become a lazy property.

Reply:
Done, also for the `LocalScanner`.